mlir/lib/Dialect/AMDGPU/IR/AMDGPUDialect.cpp

```diff
@@ -43,6 +43,21 @@ void AMDGPUDialect::initialize() {
       >();
 }
 
+bool amdgpu::AMDGPUDialect::isSharedMemoryAddressSpace(Attribute memorySpace) {
+  if (!memorySpace)
+    return false;
+  if (auto intAttr = llvm::dyn_cast<IntegerAttr>(memorySpace))
+    return intAttr.getInt() == AMDGPUDialect::kSharedMemoryAddressSpace;
```

Comment:
My understanding is that the convention is to keep the shared memory address space general for as long as possible, which to me seems like ROCDL would be the expected translation point. Can you elaborate on the reason to add an earlier dialect specific value for this?

(not necessarily a direct stakeholder here, mainly concerned about how magic memory space constants can be difficult to manage).

Reply:
Yeah, but this is, I think, designed to not break in the case where the shared memory's already been converted.

_That_ being said, I'd argue for the value of `3` to be over in rocdl, and for his "is it shared memory" thing being a `GPUDialect` utility

Reply:
+1 that's how I'm interpreting this as well. Using the `GPUDialect` utility for as long as possible seems consistent with other dialects and IMHO is more readable. It would be good to understand what cases we're hitting that has this early conversion of the shared memory space and whether this is the right level to handle it.

Reply:
Thanks forthe comments, based on the that and some discussions, it will be better bring a new PR that includes swizzling implementation as well.